ee/server/service/maintained_apps.go

```diff
@@ -115,6 +115,15 @@ func (svc *Service) AddFleetMaintainedApp(
 		maintainedAppID = nil // don't set app as maintained if scripts have been modified
 	}
 
+	// For platforms other than macOS, installer name has to match what we see in software inventory,
+	// so we have the UniqueIdentifier field to indicate what that should be (independent of the name we
+	// display when listing the FMA). For macOS, unique identifier is bundle name, and we use bundle
+	// identifier to link installers with inventory, so we set the name to the FMA's display name instead.
+	appName := app.UniqueIdentifier
+	if app.Platform == "darwin" || appName == "" {
+		appName = app.Name
+	}
+
 	payload := &fleet.UploadSoftwareInstallerPayload{
 		InstallerFile:         installerTFR,
 		Title:                 app.Name,
```

Comment:
@iansltx should this be `appName`?

Reply:
...yep. Naturally got distracted in the middle of writing this 🙃